### PR TITLE
Coverage Improvement 

### PR DESCRIPTION
**Pull request by deepweeb**
#207
# Introduction
- All the getters are taken away from API class. These are unnecessary.
- BlockChain package coverage is improved thanks to @ymte 
- Some API tests are changed so it tests the API class better
- The test coverage is improved
# Purpose
-To finalize the application code
- Removed ReadCrawlerBlockQueryTest.java because it keeps making the build fail in windows (thanks @ashay123).

# Details
None

# Future Applicability
None

### CODE REVIEW

File: app/src/main/java/nl/tudelft/bbw/BlockChainAPI.java [hunk 1]
```diff
@@ -22,133 +21,103 @@
...
+     * All the global variables of the BlockChainAPI
+     */
     private static BlockController blockController;
     private static BlockVerificationController blockVerificationController;
     private static User owner;
 
     /**
-     * Private constructor to make sure that the class cannot be initialized
-     */
-    private BlockChainAPI() {
```
Comment (humanjas) [inline app/src/main/java/nl/tudelft/bbw/BlockChainAPI.java]: This method should not be removed.

File: app/src/test/java/nl/tudelft/bbw/blockchain/UserTest.java [hunk 2]
```diff
@@ -69,4 +71,45 @@ public void getsetPrivateKey() throws Exception {
...
+        assertFalse(alice.equals(bob));
+    }
+
+    /**
+     * Test if different named users are not the same
+     */
+    @Test
+    public void testNotEqualsName() {
+        User alice = new User("Alice", "ibanA", ED25519.getPublicKey(ED25519.generatePrivateKey()));
+        User bob = new User("Bob", "ibanA", ED25519.getPublicKey(ED25519.generatePrivateKey()));
```
Comment (ymte) [inline app/src/test/java/nl/tudelft/bbw/blockchain/UserTest.java]: Good tests.

File: app/src/main/java/nl/tudelft/bbw/BlockChainAPI.java [hunk 3]
```diff
@@ -22,133 +21,103 @@
...
+            throws HashException, BlockAlreadyExistsException, NoSuchAlgorithmException,
+            InvalidKeyException, SignatureException {
         try {
             final byte[] message = user.getPublicKey().getEncoded();
             final byte[] signature = ED25519.generateSignature(message, owner.getPrivateKey());
-            //Adding the user into your own chain
             return blockController.addBlockToChain(user, signature, message);
         } catch (DatabaseException e) {
-            return null;
+            e.printStackTrace();
```
Comment (humanjas) [inline app/src/main/java/nl/tudelft/bbw/BlockChainAPI.java]: Remove this and the next line, just add the DatabaseException to the throws from the method

**Review by ghost — APPROVED** [review]

**Review by tjappo — CHANGES_REQUESTED** [review]
Have a look at the requested changes

File: app/src/main/java/nl/tudelft/bbw/BlockChainAPI.java (hunk 3 above)
Comment (deepweeb) [inline app/src/main/java/nl/tudelft/bbw/BlockChainAPI.java]: This is spotted. Thanks

**Review by tjappo — APPROVED** [review]

**Review by ashay123 — APPROVED** [review]

**Comment by ashay123** [comment]
Why did you delete the test? Doesn't this worsen the coverage?

**Comment by deepweeb** [comment]
It does but this keeps the build failing in window

**Review by nzarin — APPROVED** [review]
Good improvement. Final version. Nice coverage.

**Comment by ymte** [comment]
The worse the coverage, the better the commit ~ L. Torvalds, 1993

**Comment by ashay123** [comment]
Well, travis uses linux, so maybe locally the build would fail, but on the server it won't.
image

**Comment by deepweeb** [comment]
These things are so tiresome to fix.

**Pull request merged**